### PR TITLE
Dupey McDupe

### LINKED ISSUE #264: v0.3.0 - Duplicate Checking not working
**Describe the bug**
Duplicate checking not working when message limit is set to 0

**To Reproduce**
Set duplicate limit to 0, duplicate time to 60. 
Submit multiple copies of the same message (more than 2) 
Only the first is filtered, subsequent messages are submitted

**Expected behavior**
All messages are filtered correctly

**Additional context**
This **MAY** have already been fixed with the commits i've made but intermittently still seeing the issue. 
Issue for visibility. 
This one should block release of 0.3.0 for now.

### PR DESCRIPTION
**Pull request by davidmckenzie**
# Description

Why waste time say lot word when few word do trick?

Fixes #264

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested a few different dupe settings, needs testing from those suffering from the issue

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated changelog.md with changes made

### CODE REVIEW

**Comment by davidmckenzie** [comment]
@DanrwAU @s3m1s0n1c you guys wanna test on this one? Simple but dodgy solution, just have an array outside of the main message processing that's length is determined by the dupelimit vars.

**Comment by DanrwAU** [comment]
Let me rollback all the rolling back i did yesterday and i'll give it a go lol.

**Comment by DanrwAU** [comment]
This seems to be tagging different capcodes as dupes.

**Comment by davidmckenzie** [comment]
Ah welp, yeah forgot we match on address as well, easy enough fix :)

**Comment by davidmckenzie** [comment]
That should do it

**Comment by DanrwAU** [comment]
That looks better i'll run it for a few hours and see how we go 😁

File: server/routes/api.js [hunk 1]
```diff
@@ -536,17 +537,52 @@ router.get('/capcodes/agency/:id', isLoggedIn, function(req, res, next) {
...
+      var timeDiff = datetime - dupeTime;
+      // if duplicate filtering is enabled, we want to populate the message buffer and check for duplicates within the limits
+      var matches = _.where(msgBuffer, {message: data.message, address: data.address});
+      if (matches.length > 0) {
+        if (dupeTime != 0) {
+          // search the matching messages and see if any match the time constrain
+          var timeFind = _.find(matches, function(msg){ return msg.datetime > timeDiff; });
+          if (timeFind) {
+            res.status(200);
+            return res.send('Ignoring duplicate');
```
Comment (DanrwAU) [inline server/routes/api.js]: Should have an info logger here too 
logger.main.info(util.format('Ignoring duplicate: %o', message));

File: server/routes/api.js [hunk 2]
```diff
@@ -536,17 +537,52 @@ router.get('/capcodes/agency/:id', isLoggedIn, function(req, res, next) {
...
+          }
+        } else {
+          // if no dupeTime then just end the search now, we have matches
+          res.status(200);
+          return res.send('Ignoring duplicate');
```
Comment (DanrwAU) [inline server/routes/api.js]: And here 
logger.main.info(util.format('Ignoring duplicate: %o', message));

Comment (davidmckenzie) [reply server/routes/api.js]: Good catch, added both :)

**Pull request merged**